### ISSUE TITLE
check the loss function

### ISSUE THREAD

**Issue by eeholmes** [SAFS-Varanasi-Internship/mindthegap#57]
make sure it only uses pixels where full_target is not nan

Add PR

**Comment by troyrusso** [comment]
@eeholmes 

Right now the model trains with plain mean squared error on the full target, every ocean pixel counted, including the visible observations it can copy and the land and gap pixels set to zero; before, it trained with a masked MSE that counted only the held-out synthetic-cloud pixels, the ones the model could not see and had to reconstruct. The masked version is the better measure of fill quality, because it strips out the trivial copying of pixels the model already has in its input, but it fails as a training loss here: scoring only the hidden pixels leaves the observed context and the estimate-flag behavior unconstrained, so the model never learns to reproduce the field or to fill on command, and at inference the gapfill relabel then produces bad fills. So, the masked loss belongs at evaluation, where isolating the hidden pixels is exactly what you want, while training uses the full target, which teaches the whole-field reproduction and flag conditioning that the gap-fill mechanism depends on.

Now (full target, plain MSE):

In the train cell, the compile:

gmodel.compile(tf.keras.optimizers.Adam(3e-4, clipnorm=1.0), loss="mse", metrics=["mae"], jit_compile=False)

In build_tile, the target:

Y = np.nan_to_num(y[d], nan=0.0).astype("float32")[..., None]   # full field, 1 channel
where y = (chl_log - mean) / std is the standardized true field (the synthetic clouds are hidden only from the input, not from this target), and nan_to_num sets land and real-gap pixels to 0.

(no custom loss function used for training, uses Keras's built-in "mse")

Before (masked MSE, held-out clouds only):


Y = np.stack([np.nan_to_num(y, nan=0.0), ch["fake_cloud_flag"]], -1)[d]   # 2 channels: [value, held-out mask]
gmodel.compile("adam", loss=mtg.masked_mse, jit_compile=False)



def masked_mse(y_true, y_pred):                 # counts error ONLY where mask == 1 (the hidden synthetic clouds)
    y = y_true[..., 0:1]; m = y_true[..., 1:2]
    return tf.reduce_sum(tf.square(y - y_pred) * m) / (tf.reduce_sum(m) + 1e-6)


Same masked idea, but where it belongs, as an eval metric at inference rather than a training loss:
held = (fake_cloud_flag > 0) & gocean & np.isfinite(truth)      # pixels hidden for the test
mae_fill = np.abs(pred[held] - truth[held]).mean()             # fill quality only, then compare to baseline
